diff --git a/lib/index.cjs b/lib/index.cjs
index 5df1cca065bc0ea42cec7f0ce15c65731f8635b0..e59be1661422dc768648ac166beaadb66e7f4d7b 100644
--- a/lib/index.cjs
+++ b/lib/index.cjs
@@ -176,7 +176,7 @@ const setupTsRunner = (workerPath, { execArgv, tsRunner }) => {
       execArgv = ["-r", pnpApiPath, ...execArgv];
       const pnpLoaderPath = path__default["default"].resolve(pnpApiPath, "../.pnp.loader.mjs");
       if (isFile(pnpLoaderPath)) {
-        execArgv = ["--experimental-loader", pnpLoaderPath, ...execArgv];
+        execArgv = ["--experimental-loader", node_url.pathToFileURL(pnpLoaderPath), ...execArgv];
       }
     }
   }
